pkgs/applications/networking/sniffers/kismet/default.nix

```diff
@@ -39,6 +39,10 @@ stdenv.mkDerivation rec {
       --replace "-m 4550" ""
   '';
 
+  preConfigure = ''
+    export PATH=${pkg-config}/bin/:$PATH
```

Comment:
This adds the `hostPlatform` `pkgs-config` to `PATH`, which (is most cases) is not runnable on `buildPlatform`.

Reply:
I just want to clarify: Are you saying this breaks most builds or that this fixes most builds? Native builds are currently fine for the existing package on both aarch64 and x86_64 (probably most builds). Cross builds are currently broken (specifically x86_64 -> aarch64, but probably all cross-builds), which is what my pull request addresses.

Reply:
This would not break native build, but still won't fix cross build. If it works for you, could be binfmt_misc interfering, please disable that and try again.

Reply:
A proper fix is to replace all hardcoded references to `pkg-config` with `$PKG_CONFIG`, which points to the pkg-config binary (something like `aarch64-unknown-linux-gnu-pkg-config`) of the correct host/target platforms.

Reply:
> This would not break native build, but still won't fix cross build. If it works for you, could be binfmt_misc interfering, please disable that and try again.

Aw, you are right. Good catch. I've been emulating pkg-config and the protoc compilers without realizing it :-|. This does not work without qemu-user-static package on Debian based machines or  boot.binfmt.emulatedSystems = [ "aarch64-linux" ] on NixOS. Still, this is the only way I have been able to semi-cross-compile on Non-NixOS machines. 

I guess if this isn't really suitable for a merge, I can create an issue, and then submit this gist: https://gist.github.com/pete3n/4d997ee43277d31e2d7ee2a987018f5f  as a solution to the issue? I just don't want other people to struggle the way I have to build this with pkgsCross.

Reply:
Sumitted an issue: https://github.com/NixOS/nixpkgs/issues/276165

That can be closed with either the provided gits or possibly a revised version of this pull request. Obviously at your discretion. I just want to provide some workarounds to my cross-build issue.